### PR TITLE
Off-session draft-order charges

### PR DESCRIPTION
**Pull request by joebon**
## Summary

**Related Issue**: polarsource/feedback#127

Adds the **off-session charge API** — charge a customer's saved payment method off-session by creating a draft order and finalizing it. Scoped to **one-time, fixed-price products**. This is **part 1 of 2** and the base branch for the stacked "Set on order" PR (#12090).

## What

- Draft order status + nullable `invoice_number` (assigned at finalize) — with migration
- `off_session_charges_enabled` organization feature flag
- `POST /v1/orders` (create draft) and finalize endpoints + service, authorized via an `OrderPolicyGuard` dependency
- Only **fixed-price** one-time products are chargeable; subscription, seat-based, and pay-what-you-want products are rejected. `OrderCreate` takes no `amount` — it's fully determined by the product's price
- Client SDK regen, mobile-app draft-order status handling, and a backoffice draft-order badge

## Why

Merchants need to charge customers off-session against a saved payment method (e.g. manual/overage charges), outside the interactive checkout flow.

## How

Draft orders are persisted with `status=draft` and no invoice number. Finalize resolves the payment method, attempts an off-session charge synchronously, and on success transitions the order to paid, assigns an invoice number, and grants benefits; on failure it reverts to draft for retry. Authorization is centralized in an `OrderPolicyGuard`.

## Checklist

- [x] This PR addresses a single concern (one bug fix, one feature, one refactor)
- [x] The diff is reasonably sized and easy to review
- [x] New functionality is covered by tests
- [x] Linting and type checking pass (`uv run task lint && uv run task lint_types`)
- [x] No unrelated changes or drive-by fixes are included

## Summary by CodeRabbit

* **New Features**
  * Draft order creation and finalize flow for off‑session charges with saved payment methods; new API endpoints to create and finalize draft orders.

* **Improvements**
  * Global "Draft" order status with gray badge/styling and list filter.
  * Order header shows ID when invoice number is missing.
  * Feature‑flag gating for off‑session charges and clearer finalize error responses.

* **Tests**
  * Expanded tests covering draft/create/finalize flows, payment edge cases, and custom‑price validation.

### CODE REVIEW

File: server/polar/order/service.py [hunk 1]
```diff
@@ -700,6 +779,383 @@ async def _create_order_from_checkout(
...
+                    {
+                        "type": "value_error",
+                        "loc": ("body", "customer_id"),
+                        "msg": "Customer does not exist.",
+                        "input": payload.customer_id,
+                    }
+                ]
+            )
+
+        currency = organization.default_presentment_currency
```
Comment (frankie567) [inline server/polar/order/service.py]: We should probably be able to pass a currency (optional) in the API input; if the merchant wants to force a specific currency.

**Comment by joebon** [comment]
@coderabbitai review

File: server/polar/product/price_set.py [hunk 2]
```diff
@@ -93,8 +98,104 @@ def get_default_price(self) -> ProductPrice:
         return self.prices[0]
 
 
+def validate_custom_price_amount(
```
Comment (frankie567) [inline server/polar/product/price_set.py]: Let's place it in a dedicated module `product/custom_price.py`

File: server/polar/order/service.py [hunk 3]
```diff
@@ -700,6 +779,383 @@ async def _create_order_from_checkout(
...
+                            "off-session charge API."
+                        ),
+                        "input": payload.product_id,
+                    }
+                ]
+            )
+
+        # Off-session charges currently support fixed-price products only: the
+        # amount must be fully determined by the product. Custom (pay-what-you-want)
+        # and free prices are rejected.
```
Comment (frankie567) [inline server/polar/order/service.py]: I think free price should be supported. Technically, we'll just need to skip the payment step and it'll all work naturally. Can help to fulfill some interesting use-cases.

**Review by frankie567 — CHANGES_REQUESTED** [review]
Great work! Some things to tweak, but it's super solid!

File: server/polar/order/service.py (hunk 3 above)
Comment (joebon) [reply server/polar/order/service.py]: Makes sense! Ok to add it as a follow-up? I also have a vague recollection of you @pieterbeulque mentioning something around free price products, or am I hallucinating?

Comment (frankie567) [reply server/polar/order/service.py]: Definitely!

**Review by frankie567 — APPROVED** [review]
